opal-core-ws/src/main/java/org/obiba/opal/web/project/ProjectTransientDatasourcesResource.java

```diff
@@ -80,10 +96,66 @@ public Response createDatasource(Magma.DatasourceFactoryDto factoryDto) {
           .entity(Dtos.asDto(ds).build()).build();
     } catch(MagmaRuntimeException e) {
       MagmaEngine.get().removeTransientDatasource(uid);
+
+      if (e instanceof DatasourceParsingException) {
+        cacheDatarsourceParseErrorLog((DatasourceParsingException) e);
+      }
```

Comment:
so for the same sessionid, before eviction time and after a import failure then a import success, one could retrieve errors (from the import failure).